runtime/manifest-parser.peg

```diff
@@ -38,6 +38,18 @@
       }
       return;
     }
+    if (result.typeKind) {
```

Comment:
This is accidentally treating `typeKind` as a generic parser level concept. The idea was that `result` forms a generic AST where nodes contain `location` & `kind`. Can we use `kind` instead of `typeKind` or set both properties?

Reply:
Sure; will update to use the existing kind (and location) fields.